docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -321,32 +333,155 @@ kind: Logstash
           requests:
             storage: 10Gi
-  config: 
-    log.level: info
-    queue.type: persisted
-    path.queue: /usr/share/logstash/pq
+  pipelines:
+    - pipeline.id: beats
+      dead_letter_queue.enable: true
+      path.dead_letter_queue: /usr/share/logstash/dlq <2>
+      config.string: |
+        input {
+          beats {
+            port => 5044
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${ECK_ES_HOSTS}" ]
+            user => "${ECK_ES_USER}"
+            password => "${ECK_ES_PASSWORD}"
+            ssl_certificate_authorities => "${ECK_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+    - pipeline.id: dlq_read
+      dead_letter_queue.enable: false
+      config.string: |
+        input {
+          dead_letter_queue {
+            path => "/usr/share/logstash/dlq" <2>
+            commit_offsets => true
+            pipeline_id => "beats"
+            clean_consumed => true
+          }
+        }
+        filter {
+          mutate {
+            remove_field => "[geoip][location]"
+          }
+        }
+        output {
+          elasticsearch {
+            hosts => [ "${ECK_ES_HOSTS}" ]
+            user => "${ECK_ES_USER}"
+            password => "${ECK_ES_PASSWORD}"
+            ssl_certificate_authorities => "${ECK_ES_SSL_CERTIFICATE_AUTHORITY}"
+          }
+        }
+----
+<1> The `name` values in the `volumeMount` for the container in the podTemplate section and the name of the `volumeClaimTemplate` must match
+<2> Set the `path.dead_letter_queue` setting in the pipeline config to match the `mountPath` in the `volumeMount` for pipelines that are writing to the Dead Letter Queue, and set the `path` setting of the `dead_letter_queue` plugin for the pipeline that will read from the Dead Letter Queue.
+
+
+[float]
+== Updating the volume claim settings
+
+Changes are currently forbidden in `spec.volumeClaimTemplates`, such as changing the storage class or volume size. To make these changes, you have to fully delete the Logstash object, delete and recreate or resize the volume and create a new Logstash object.
+
+Before deleting or resizing a persistent queue volume, ensure that the queue is empty. When using the persistent queue, we recommend setting `queue.drain: true` on the Logstash Pods to ensure that the persistent queue is drained when Pods are shutdown. In conjunction with this, note that you should also increase the `terminationGracePeriodSeconds` to a large enough value to drain your queue.
+
+The following example shows how to configure a Logstash resource to drain the queue and increase the termination grace period:
+
+[source,yaml,subs="attributes,+macros,callouts"]
+----
+apiVersion: logstash.k8s.elastic.co/v1alpha1
+kind: Logstash
+metadata:
+  name: logstash
+spec:
+  # some configuration attributes omitted for brevity here
+  config:
+    queue.drain: true
+  podTemplate:
+    spec:
+      terminationGracePeriodSeconds: 604800
```

Comment:
168h seems a lot? and we say below that it can be an issue.
```suggestion
      terminationGracePeriodSeconds: 600
```

Reply:
This comes from the [Logstash docs](https://github.com/elastic/logstash/blob/3d257083c60bf80852e68304b10746ca34b279bb/docsk8s/troubleshooting/ls-k8s-troubleshooting-methods.asciidoc#draining-the-pq), where we state that we should set an artificially high `terminationGracePeriodSeconds` to drain the queue. The intention was to set the high value that we would prefer, but caution that it may not be honored